hnn_core/viz.py

```diff
@@ -543,17 +560,20 @@ def plot_spikes_raster(cell_response, trial_idx=None, ax=None, show=True):
     spike_gids = np.concatenate(
         np.array(cell_response._spike_gids, dtype=object)[trial_idx])
 
-    cell_types = ['L2_basket', 'L2_pyramidal', 'L5_basket', 'L5_pyramidal']
-    cell_type_colors = {'L5_pyramidal': 'r', 'L5_basket': 'b',
-                        'L2_pyramidal': 'g', 'L2_basket': 'w'}
-
     if ax is None:
         _, ax = plt.subplots(1, 1, constrained_layout=True)
 
+    # Check if custom colors are provided, else use the default color cycle
+    if colors is None:
+        colors = plt.rcParams['axes.prop_cycle'].by_key()['color']
+    color_iter = iter(colors)
+
     events = []
     for cell_type in cell_types:
         cell_type_gids = np.unique(spike_gids[spike_types == cell_type])
         cell_type_times, cell_type_ypos = [], []
+        color = next(color_iter)
+
         for gid in cell_type_gids:
             gid_time = spike_times[spike_gids == gid]
             cell_type_times.append(gid_time)
```

Comment:
while you are at it, I am wondering if this could be addressed as well. The following line:

```py
cell_type_ypos.append(-gid)
```

causes cells that spike with neighboring gids to overlap. I have been staring at these raster plots recently and it's very hard to tell how many times the same cell spiked (important to understand the underlying dynamics). Adding a small offset between nearby cells should address that problem

Reply:
I'm okay with the spikes looking like dots ... it's just a function of the number of cells in our network. Did a quick google image search of "spike raster plot" and the plots do look dotted when there are more neurons. I guess the y-offset = -gid is helpful since it allows you to identify the cell, so maybe best not to touch that. @ntolley any opinion here?

Reply:
Yesterday, both @ntolley and I had a brief look at the "dot" version and, IIRC, we both agreed it looks good.